src/utils/callcallback.js

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2018 ml5
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+export default function callCallback(promise, callback) {
```

Comment:
I'm not sold on `callCallback` - that requires handling a promise internally within the `async` functions - something like `return callCallback(Promise.resolve(result), callback);` could theoretically be used to return a result that is non-promised but I think you lose a lot of the wonder of an `async` function by doing this. As I suggested in #143 a wrapping function that creates callbacks from the `async` Promises would probably be my preferred technique - I have a working suggestion that's updated now here for decorators: https://github.com/ml5js/ml5-library/compare/master...meiamsome:testing/promise-to-callback-decorator or here for with class properties: https://github.com/ml5js/ml5-library/compare/master...meiamsome:testing/promise-to-callback-class-properties, although both of them need to used babel-eslint because the features are not fully standardised they both give the benefits that the functionality of the Promise and the callback(s) are intrinsically linked. Obviously, if supporting one callback type is required it would significantly reduce the complexity of the wrapping fn/decorator.

Reply:
Is there any specific behavior in the exposed API surface, in this branch, that seems incorrect? I'm finding it hard to parse from your comment whether the issue is code aesthetics or if there is a specific method behavior that would be different if we were to go with a different approach.

Reply:
I have gone over the PR again and am fairly sure that the exposed* API is mostly correct - my main concern is an aesthetic one here, but there are some other important things such as the fact any errors inside the functions are just never sent to the callback function at the moment, and adding it in is a matter of wrapping almost every function in a try/catch statement to send it correctly at that least, which is a lot of boilerplate when that could be handled by wrapping the function itself. Also, the if/else if/.../else chaining for parsing arguments that's just a big mess. 

* In addition, there's some `async` functions that don't allow callbacks but I'm not sure these are public or private (see the other discussion on private properties for that, I guess)